server/models/Profile.js

```diff
@@ -0,0 +1,50 @@
+const mongoose = require("mongoose");
+
+const profileSchema = new mongoose.Schema({
+  firstName: {
+    type: String,
+    required: true,
+  },
+  lastName: {
+    type: String,
+    required: true,
+  },
+  gender: {
+    type: String,
+    enum: ['MALE', 'FEMALE', 'OTHER'],
+    required: true,
+    default: 'MALE'
+  },
+  birthDate: {
+    type: Date,
+    required: false
+  },
+  email: {
+    type: String,
+    required: true,
+    unique: true
+  },
+  phoneNumber: {
+    type: Number,
+    required: true
+  },
+  location: {
+    type: String,
+    required: true
+  },
+  description: {
+    type: String,
+    required: true
+  },
+  availability: [{
+    type: Date, 
+    required: true
+  }],
```

Comment:
Could you elaborate on the decision here a bit more? An array of dates, thinking of the front-end here. Would this mean the user needs to input every day they are available. 

Here are my thoughts. What if we make a general availability `Mon, Tues, Wed, Thurs, Fri, etc` and allow the user to select which days they are available as well as a time range of availability. UI/UX wise this is a lot more efficient. Thoughts?

Reply:
Sounds good Sir. I was actually thinking of making it in such a way that we can receive the availability through the HTML Datetime input and store in the availability field and extracting the date and time, but I get your point (we should be working with' week days' and not 'dates'. So I will the change the field to an array of objects, with `day` and `time range` properties?